crowbar_framework/spec/lib/crowbar/lock/shared_non_blocking_spec.rb

```diff
@@ -0,0 +1,177 @@
+require "spec_helper"
+require "timeout"
+require "rspec/expectations"
+
+RSpec::Matchers.define :be_unlocked do
+  match do |actual|
+    !actual.locked? && !File.exists?(actual.path)
+  end
+  failure_message do |actual|
+    actual.locked? ?
+      "expected that #{actual} would not be locked"
+    : "expected that #{actual.path} would not exist"
+  end
+end
+
+RSpec::Matchers.define :be_locked_by_me do
+  match do |actual|
+    actual.locked? && File.exists?(actual.path)
+  end
+  failure_message do |actual|
+    ! actual.locked? ?
```

Comment:
Style/MultilineTernaryOperator: Avoid multi-line ?: (the ternary operator); use `if`/`unless` instead.<br>Style/SpaceAfterNot: Do not leave space between `!` and its argument.